sycl/source/detail/graph/graph_impl.cpp

```diff
@@ -165,35 +172,6 @@ bool isPartitionRoot(std::shared_ptr<node_impl> Node) {
   }
   return true;
 }
-
```

Comment:
Good job, I think the changes in this refactor makes the code significantly cleaner :+1: 

However, we are still mixing `impl`  (e.g. exec_graph_impl / node_impl) object implementations and user facing object implementations (e.g. node / executable_command_graph) in the same files (i.e., graph_impl.cpp, node_impl.cpp, etc.).

I remember this was quite confusing when I was initially learning the codebase.  I wonder if it would make sense to split things further into `node.cpp`, `command_graph.cpp`, etc. which would live under `sycl/source/graph`

This would be similar to other sycl classes. For example, there is `queue.cpp` and `queue_impl.cpp`, `accessor.cpp` and `accessor_impl.cpp`, etc.

Reply:
I was also thinking there is some more potential work to be done here to separate the code from `detail` namespace to correct impl files. As agreed there we'll do a follow up work in the next PR.